### ISSUE TITLE
Font Library: add terms and conditions

### ISSUE THREAD

**Issue by matiasbenedetto** [WordPress/gutenberg#58774]
## What problem does this address?
- Give users a way to accept or reject loading the data from a registered font collection.
- Give font collection creators the ability to add terms and conditions text to inform users about that. 

    
## What is your proposed solution?
- Add terms and conditions field in the font collection.
- Render the terms and conditions from a font collection in the frontend. 
- All the font collections need users' permission to load the assets.
- Make the [Google fonts permission dialog](https://github.com/WordPress/gutenberg/issues/53882#issuecomment-1692936996) font collection agnostic.

--- 
Core trac ticket: https://core.trac.wordpress.org/ticket/60593

**Comment by getdave** [comment]
@matiasbenedetto If WordPress (e.g. Google Fonts) or Themes are bundling Font Families, it would be good to allow users to:

- be informed about the conditions of usage
- reject those Collections

I believe blocking registration of fonts can be done via filters so technically it is possible but requires technical knowledge.

How difficult is it going to be to add a UI to allow users to revoke collections? Have we made any plans for implementing this during the 6.5 cycle?

@fabiankaegy @colorful-tones @youknowriad For your thoughts also. We are entering Beta 1, but I _think_ if this is deemed sufficiently important it's passable to allow the work to be completed during the Beta phase.

**Comment by youknowriad** [comment]
I think the UI is already in place, you have to accept the license the first time you want to use google fonts (first time you open the tab) and even after than, you have to "install" fonts to work.

**Comment by getdave** [comment]
I'm wondering whether the behaviour you describe is specific to Google Fonts?

**Comment by matiasbenedetto** [comment]
The ask for permission and explanation about the behavior is already in place, but that's specific to the Google Fonts collection. In this ticket, I think we should discuss if we need to generalize by adding the ability to the font collection to provide texts with their license or similar.

**Comment by youknowriad** [comment]
I think it would be good to generalize indeed but I don't believe it's a 6.5 requirement.

**Comment by colorful-tones** [comment]
> I think it would be good to generalize indeed but I don't believe it's a 6.5 requirement.

Yeah, totally agree. Not a 6.5 requirement. However, I would love to see the addition of a UI to allow users to revoke a Collections acceptance (Local Storage) for 6.5. That is captured in https://github.com/WordPress/gutenberg/issues/54826 , and yes it would be nice to see it implemented in a generalized nature so that other potential collections could leverage the same method.
